src/backend/fs.rs

```diff
@@ -255,73 +256,65 @@ pub fn read_catalog(file_hash: &Blake3Hash) -> Result<Vec<BaoHash>> {
     Ok(bao_hashes)
 }
 
-#[allow(unused_variables)]
-pub fn delete_file(pk: Secp256k1PubKey, file_bytes: &[u8]) -> Result<()> {
-    let pk_bytes = pk.to_bytes();
-    let (x_only_pk, _) = pk.into_inner().x_only_public_key();
-
-    let file_hash = Blake3Hash(blake3::keyed_hash(&x_only_pk.serialize(), file_bytes));
-    trace!(">>>>>file_hash::   {}", file_hash);
-
-    for vol in &SYS_CFG.volumes {
-        let seg_file = &vol.path.join(SEGMENT_DIR).join(file_hash.to_string());
-        let seg_dir = &vol.path.join(SEGMENT_DIR);
-        remove_dir_contents(seg_dir, seg_file.to_path_buf()).unwrap();
-    }
-
-    for vol in &SYS_CFG.volumes {
-        let cat_path = &vol.path.join(CATALOG_DIR).join(file_hash.to_string());
-        let cat = &vol.path.join(CATALOG_DIR);
-        remove_dir_catalogs(cat.to_path_buf(), cat_path.to_path_buf()).unwrap();
-    }
-    Ok(())
+pub async fn read_slices(
+    //hash: bao::Hash, // from Blake3
+    file_bytes: &[u8],
+    //slice_start: u64, //let slice_start = 65536;
+    //slice_len: u16,   // let slice_len = 8192;
+) -> Result<Vec<u8>> {
```

Comment:
It makes no sense to provide the entire file to this method. This is supposed to be for retrieving just a slice of the file. It needs to return the range of the raw bytes on disk. It can make the disk IO calls by reading the catalog and then the appropriate segment.

I've updated the issue so hopefully the task is more clear:
https://github.com/diba-io/carbonado-node/issues/18

Reply:
those values where for initial testing only.
actual slice data comes from the params.